test/NVWS/insert_aref.mlir

```diff
@@ -0,0 +1,133 @@
+    "use"(%result_0) : (tensor<128x128xf32, #blocked>) -> ()
+    tt.return
+  }
+
+  // CHECK-LABEL: @specialize_load_only
+  tt.func @specialize_load_only(%arg0: !tt.tensordesc<tensor<128x64xf16, #shared>>, %arg1: i32) {
+    %c0_i32 = arith.constant 0 : i32
+    %c1_i32 = arith.constant 1 : i32
+    scf.for %arg2 = %c0_i32 to %arg1 step %c1_i32  : i32 {
+      // CHECK: nvws.aref.put.enter
+      // CHECK: nvws.descriptor_load
+      // CHECK: nvws.aref.put.exit
+      %0 = tt.descriptor_load %arg0[%arg2, %arg2] {ttg.partition = 2 : i32} : !tt.tensordesc<tensor<128x64xf16, #shared>> -> tensor<128x64xf16, #blocked1>
+      // CHECK: nvws.aref.get.enter
+      // CHECK: [[REG:%.*]] = ttg.local_load
+      // CHECK: nvws.aref.get.exit {{.*}} [#nvws.async_op<none>]
+      // CHECK: "use"([[REG]])
+      "use"(%0) {ttg.partition = 0 : i32} : (tensor<128x64xf16, #blocked1>) -> ()
+    } {tt.warp_specialize, ttg.partition.stages = [0 : i32, 1 : i32, 0 : i32]}
+    tt.return
+  }
+
+  // CHECK-LABEL: @no_value_aref
+  tt.func @no_value_aref(%arg0: tensor<128x64xf16, #blocked1>, %arg1: i32) {
+    %c0_i32 = arith.constant 0 : i32
+    %c1_i32 = arith.constant 1 : i32
+    // CHECK-NOT: nvws.aref.create
+    scf.for %arg2 = %c0_i32 to %arg1 step %c1_i32  : i32 {
+      %0 = "producer"(%arg0, %arg2) {ttg.partition = 1 : i32} : (tensor<128x64xf16, #blocked1>, i32) -> tensor<128x64xf16, #blocked1>
+      "use"(%0) {ttg.partition = 0 : i32} : (tensor<128x64xf16, #blocked1>) -> ()
+    } {tt.warp_specialize, ttg.partition.stages = [0 : i32, 1 : i32, 0 : i32]}
+    tt.return
+  }
+
+  // CHECK-LABEL: @load_used_as_reg_and_smem
+  tt.func @load_used_as_reg_and_smem(%arg0: !tt.tensordesc<tensor<128x64xf16, #shared>>, %arg1: i32) {
+    %c0_i32 = arith.constant 0 : i32
+    %c1_i32 = arith.constant 1 : i32
+    scf.for %arg2 = %c0_i32 to %arg1 step %c1_i32  : i32 {
+      // CHECK: nvws.aref.put.enter
+      // CHECK: nvws.descriptor_load
+      // CHECK: nvws.aref.put.exit
+      %0 = tt.descriptor_load %arg0[%arg2, %arg2] {ttg.partition = 2 : i32} : !tt.tensordesc<tensor<128x64xf16, #shared>> -> tensor<128x64xf16, #blocked1>
+      %alloc = ttg.local_alloc %0 {ttg.partition = 2 : i32} : (tensor<128x64xf16, #blocked1>) -> !ttg.memdesc<128x64xf16, #shared, #smem>
+      // CHECK-DAG: [[GET_BUF1:%.*]] = nvws.aref.get.enter {{.*}} {aref_tag = "aref_0", ttg.partition = 0 : i32}
+      // CHECK-DAG: [[GET_BUF2:%.*]] = nvws.aref.get.enter {{.*}} {aref_tag = "aref_0", ttg.partition = 1 : i32}
+      // CHECK-DAG: [[REG:%.*]] = ttg.local_load [[GET_BUF1]]
+      // CHECK-DAG: nvws.aref.get.exit {{.*}} [#nvws.async_op<none>] {aref_tag = "aref_0", ttg.partition = 0 : i32}
+      // CHECK: "use1"([[REG]])
+      // CHECK: "use2"([[GET_BUF2]])
+      // CHECK: nvws.aref.get.exit {{.*}} [#nvws.async_op<none>] {aref_tag = "aref_0", ttg.partition = 1 : i32}
```

Comment:
@3gx I believe this is the first time we are seeing multiple get_exits on one producer. The following logic in LowerRef is broken, I think we need to sum the arrival counts over all get_exits in all partitions.

https://github.com/triton-lang/triton/blob/main/third_party/nvidia/lib/Dialect/NVWS/Transforms/LowerAref.cpp#L147-L149

Reply:
there are also [lit-tests ](https://github.com/triton-lang/triton/blob/main/test/TritonGPU/rewrite-partition-dependencies.mlir#L11)for `rewrite-partition-dependencies`  pass.
This PR https://github.com/triton-lang/triton/pull/7648  adds `lower-aref` support for multiple consumers